engine/source/output/restart/rdresb.F

```diff
@@ -1952,13 +1952,16 @@ SUBROUTINE FVRREST(MONVOL)
        
        !---AIRBAGS_NODE_ID_SHIFT
        ! first node identifier used to plot polyhedron centroids is 1+AIRBAGS_NODE_ID_SHIFT
-       TMP = NUMNOD
-       AIRBAGS_NODE_ID_SHIFT = 1 + 1+FLOOR(LOG10(TMP))    !first index for virtual node identifiers (next available power of 10)
-       IF (AIRBAGS_NODE_ID_SHIFT > 10) THEN                             ! example :                  NUMNOD =   123,456
-                                                                        !   ->        AIRBAGS_NODE_ID_SHIFT = 1,000,000
-          WRITE(ISTDO,'(A)')" WARNING FVMBAG VISUALIZATION : INDEX SHIFT IS MORE THAN 10 DIGITS"
-          WRITE(IOUT ,'(A)')" WARNING FVMBAG VISUALIZATION : INDEX SHIFT IS MORE THAN 10 DIGITS" 
+       AIRBAGS_NODE_ID_SHIFT = 0
+       IF(NUMNOD > 0)THEN
+         TMP = NUMNOD
+         AIRBAGS_NODE_ID_SHIFT = 1 + 1+FLOOR(LOG10(TMP))    !first index for virtual node identifiers (next available power of 10)
```

Comment:
Maybe it should be 1 + FLOOR(LOG10)) instead of "1+1+FLOOR" ? 

Also, I would suggest to perform the  'AIRBAGS_NODE_ID_SHIFT = 10**AIRBAGS_NODE_ID_SHIFT' if AIRBAGS_NODE_ID_SHIFT <= 9, in order to avoid integer overflows.

Reply:
Good catch ! thank you for this review. 
Yes it can be cleaned like that now : it was initially a number of digits. Not a directly a power of ten.
(So you are right I will also compare it to 9 instead of 10 : 10*9 has 10 digits)

Reply:
Hi @laurent-altr , I finally changed this calculation. I introduced a local data structure in H3D to manage identifier for virtual nodes. Some are created from /RWALL option, other ones from /MONVOL/FVMBAG* options. 
Identifiers are now adjacent and identifiers starts directly from MAX_NODE_ID +1.
It has been the case so far with /RWALL (with generates virtual nodes to plot rigid wall in h3d).

I also introduced a check (a posteriori) which check If we exceeded the maximal integer value.